pkg/cloudprovider/aws/fleet/launchtemplate.go

```diff
@@ -142,27 +143,33 @@ func (p *LaunchTemplateProvider) getSecurityGroupIds(ctx context.Context, cluste
 	return securityGroupIds, nil
 }
 
-func (p *LaunchTemplateProvider) getAMIID(context context.Context) (*string, error) {
+func (p *LaunchTemplateProvider) getAMIID(ctx context.Context) (*string, error) {
 	version, err := p.kubeServerVersion()
 	if err != nil {
 		return nil, fmt.Errorf("kube server version, %w", err)
 	}
-	paramOutput, err := p.ssm.GetParameter(&ssm.GetParameterInput{
-		Name: aws.String(fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2/recommended", version)),
+	paramOutput, err := p.ssm.GetParameterWithContext(ctx, &ssm.GetParameterInput{
+		Name: aws.String(fmt.Sprintf("/aws/service/bottlerocket/aws-k8s-%s/x86_64/latest/image_id", version)),
 	})
 	if err != nil {
 		return nil, fmt.Errorf("ssm get parameter, %w", err)
 	}
-	output := struct {
-		ImageID string `json:"image_id"`
-	}{}
-	if err := json.Unmarshal([]byte(*paramOutput.Parameter.Value), &output); err != nil {
-		return nil, fmt.Errorf("unmarshal parameter output, %w", err)
+	return paramOutput.Parameter.Value, nil
+}
+
+func (p *LaunchTemplateProvider) getUserData(cluster *v1alpha1.ClusterSpec) (*string, error) {
+	t := template.Must(template.New("bottlerocketUserData").Parse(bottleRocketUserData))
+	var userData bytes.Buffer
+	if err := t.Execute(&userData, cluster); err != nil {
```

Comment:
It's a bit odd to use template.Must, which crashes on failure, but then bubble up an error on t.Execute. As long as we control all sides of the template, I'm happy with just crashing because it means that something is really quite unexpected.

Reply:
I'm pretty sure `t.Execute` could fail since it depends on user-supplied data? Whereas `template.Must` really should never fail since it's 100% under our control (I assume it'd only fail if we had an invalid template like `"foo{{.Bar}"`?

Reply:
We (will) control all of these variables using validation. They can be assumed as invariants. Not too passionate about this one, but we do control it and it can be simplified (slightly).

Reply:
I don't really like having explicit `panic` in code, so not sure that's a simplification - I'd propose either keep as is or else make them both return errors.